manifests/plugin/container_gateway.pp

```diff
@@ -22,13 +36,36 @@
   Boolean $enabled = true,
   Foreman_proxy::ListenOn $listen_on = 'https',
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
+  Enum['postgres', 'sqlite'] $database_backend = 'postgres',
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Boolean $manage_postgresql = true,
+  Optional[Stdlib::Host] $postgresql_host = undef,
+  Optional[Stdlib::Port] $postgresql_port = undef,
+  String $postgresql_database = 'container_gateway',
+  String $postgresql_user = pick($foreman_proxy::globals::user, 'foreman-proxy'),
```

Comment:
I'd prefer to leave it out by default
```suggestion
Optional[String[1]] $postgresql_user = undef,
```

Then later in the body you can determine it with `pick($postgresql_user, $foreman_proxy::user)`. I'd assign it to a variable, so like `$_postgresql_user = pick($postgresql_user, $foreman_proxy::user)`.